notebooks/gnn/pyg_hetero_mag_cgs.ipynb

```diff
@@ -0,0 +1,302 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "loading extensions from /work/cugraph/notebooks/gnn/cgs_creation_extensions\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "['/work/cugraph/notebooks/gnn/cgs_creation_extensions/cgs_mag_extension.py']"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import pathlib\n",
+    "import os\n",
+    "from cugraph_service_client.client import CugraphServiceClient\n",
+    "# Create a new client instance\n",
+    "client = CugraphServiceClient()\n",
+    "\n",
+    "# Set up the creation extensions\n",
+    "ext_path = os.path.join(\n",
+    "    pathlib.Path('__file__').parent.resolve(),\n",
+    "    'cgs_creation_extensions'\n",
+    ")\n",
+    "print(f'loading extensions from {ext_path}')\n",
+    "client.load_graph_creation_extensions(str(ext_path))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from cugraph_service_client.client import RemoteGraph\n",
+    "\n",
+    "# This line may take a while if the data has not yet been downloaded.\n",
+    "graph_id = client.call_graph_creation_extension('create_mag')\n",
+    "\n",
+    "pG = RemoteGraph(client, graph_id)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Construct a Graph Store, Feature Store, and Loaders"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from cugraph.experimental.pyg_extensions import to_pyg\n",
+    "\n",
+    "feature_store, graph_store = to_pyg(pG)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from cugraph.experimental.pyg_extensions import CuGraphSampler\n",
+    "sampler = CuGraphSampler(\n",
+    "    data=(feature_store, graph_store),\n",
+    "    shuffle=True,\n",
+    "    num_neighbors=[10,25],\n",
+    "    batch_size=50,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from torch_geometric.loader import NodeLoader\n",
+    "loader = NodeLoader(\n",
+    "    data=(feature_store, graph_store),\n",
+    "    shuffle=True,\n",
+    "    batch_size=50,\n",
+    "    node_sampler=sampler,\n",
+    "    input_nodes=('author', graph_store.get_vertex_index('author'))\n",
+    ")\n",
+    "\n",
+    "test_loader = NodeLoader(\n",
+    "    data=(feature_store, graph_store),\n",
+    "    shuffle=True,\n",
+    "    batch_size=50,\n",
+    "    node_sampler=sampler,\n",
+    "    input_nodes=('author', graph_store.get_vertex_index('author'))\n",
+    ")\n"
+   ]
```

Comment:
**Optional for now** 
Is the rest of code similar to `pyg_hetero_mag_cgs` ? If yes, i think it may make sense to put this in a model.py or something and just import for both notebooks ? 

I am not reviewing this assuming it  is the same code. Please let me know if it is not.  We can probably do this later on in a followup too as this PR seems to unblock the CI.

Reply:
Code is similar but we may change the workflow a bit in the future.  Also because this is primarily for learning, the whole code should be in this file, I think

Reply:
Okay , then let it be as is.   Code duplication from a learning perspective makes sense.